internal/pkg/repository/repository.go

```diff
@@ -0,0 +1,74 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package repository provides support for building and pushing images to a repository.
+package repository
+
+import (
+    "fmt"
+    "github.com/aws/copilot-cli/internal/pkg/aws/ecr"
+)
+
+// DockerService provides support for logging in to repositories, building images and pushing images to repositories.
+type ContainerManager interface {
+    Build(uri, path, imageTag string, additionalTags ...string) error
+    Login(uri, username, password string) error
+    Push(uri, imageTag string, additionalTags ...string) error
+}
+
+// RepositoryGetter gets information of ECR repositories.
+type ECRRepositoryGetter interface {
+    GetRepository(name string) (string, error)
+    GetECRAuth() (ecr.Auth, error)
+}
+
+// ECRRepository builds and pushes images to an ECR repository.
+type ECRRepository struct {
+    repositoryName string
+
+    repositoryGetter ECRRepositoryGetter
+    docker           ContainerManager
+
+    uri string
+}
+
+// NewECRRepository instantiates a new ECRRepository.
+func NewECRRepository(name string, repositoryGetter ECRRepositoryGetter, docker ContainerManager) (*ECRRepository, error){
```

Comment:
`func New(name string, registry Registry, containerRuntime ContainerLoginBuildPusher) (*Repository, error)`

Reply:
Actually I think `containerRuntime` should not be a field of Repository as they are unrelated, instead we can make it an input to the `BuildAndPush` method